src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/Providers/AzureFilesPsBackupProvider.cs

```diff
@@ -419,12 +419,66 @@ public ProtectedItemResource GetProtectedItem()
 
         public RecoveryPointBase GetRecoveryPointDetails()
         {
-            throw new NotImplementedException();
+            string vaultName = (string)ProviderData[VaultParams.VaultName];
+            string resourceGroupName = (string)ProviderData[VaultParams.ResourceGroupName];
+            AzureFileShareItem item = ProviderData[RecoveryPointParams.Item]
+                as AzureFileShareItem;
+
+            string recoveryPointId = ProviderData[RecoveryPointParams.RecoveryPointId].ToString();
+
+            Dictionary<UriEnums, string> uriDict = HelperUtils.ParseUri(item.Id);
+            string containerUri = HelperUtils.GetContainerUri(uriDict, item.Id);
+            string protectedItemName = HelperUtils.GetProtectedItemUri(uriDict, item.Id);
+
+            var rpResponse = ServiceClientAdapter.GetRecoveryPointDetails(
+                containerUri,
+                protectedItemName,
+                recoveryPointId,
+                vaultName: vaultName,
+                resourceGroupName: resourceGroupName);
+
+            var rp = RecoveryPointConversions.GetPSAzureRecoveryPoints(rpResponse, item);
+
+            return rp;
         }
 
         public List<RecoveryPointBase> ListRecoveryPoints()
         {
-            throw new NotImplementedException();
+            string vaultName = (string)ProviderData[VaultParams.VaultName];
+            string resourceGroupName = (string)ProviderData[VaultParams.ResourceGroupName];
+            DateTime startDate = (DateTime)(ProviderData[RecoveryPointParams.StartDate]);
+            DateTime endDate = (DateTime)(ProviderData[RecoveryPointParams.EndDate]);
+
+            AzureFileShareItem item = ProviderData[RecoveryPointParams.Item]
```

Comment:
Apart from this line, the rest of this method looks very generic. Can we refactor this between Azure VM and Azure Files using the provider helper?